### PR TITLE
Drop OS X 10.7 and Growl support

### PR DESCRIPTION
**Pull request by hrubymar10**
On more test running now and also wait for Travis

### CODE REVIEW

File: src/build_info.cpp [hunk 1]
```diff
@@ -298,17 +294,9 @@ version_table_manager::version_table_manager()
 #endif
 
 #ifdef __APPLE__
-
+    // Always compiled in.
```
Comment (jyrkive) [inline src/build_info.cpp]: Simply remove this from the feature list, then. It's intended for optional features, not mandatory ones.

File: src/desktop/apple_notification.mm [hunk 2]
```diff
@@ -47,40 +32,18 @@ bool available() {
...
 }
 
-#ifdef HAVE_NS_USER_NOTIFICATION
 void send_cocoa_notification(const std::string& owner, const std::string& message);
-#endif
-#ifdef HAVE_GROWL
-void send_growl_notification(const std::string& owner, const std::string& message, const desktop::notifications::type note_type);
-#endif
 
-#pragma clang diagnostic push
```
Comment (jyrkive) [inline src/desktop/apple_notification.mm]: Don't remove this `#pragma`, or the `pop` lower down.

File: src/build_info.cpp (hunk 1 above)
Comment (hrubymar10) [reply src/build_info.cpp]: Windows has it same way, isn't it?

File: src/desktop/apple_notification.mm (hunk 2 above)
Comment (hrubymar10) [reply src/desktop/apple_notification.mm]: Sure

File: src/build_info.cpp (hunk 1 above)
Comment (jyrkive) [reply src/build_info.cpp]: Ah, you're right. OK, it's better to keep Cocoa notifications in the list.

File: src/desktop/apple_notification.mm [hunk 3]
```diff
@@ -44,6 +45,7 @@ void send_notification(const std::string& owner, const std::string& message, con
     }
 }
 
+#pragma clang diagnostic pop
```
Comment (jyrkive) [inline src/desktop/apple_notification.mm]: Nitpicking: this should be one line higher (attached to `send_notification()`, not `send_cocoa_notification()`).

**Comment by hrubymar10** [comment]
If will @jyrkive , then it it ready to merge now

**Comment by jyrkive** [comment]
Done. Thanks for the PR. 👍

File: src/desktop/apple_notification.mm [hunk 4]
```diff
@@ -47,40 +32,20 @@ bool available() {
...
-void send_growl_notification(const std::string& owner, const std::string& message, const desktop::notifications::type note_type);
-#endif
 
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wunused-parameter"
-#ifdef HAVE_NS_USER_NOTIFICATION
 void send_notification(const std::string& owner, const std::string& message, const desktop::notifications::type note_type) {
     @autoreleasepool {
         Class appleNotificationClass = NSClassFromString(@"NSUserNotificationCenter");
         if (appleNotificationClass) {
```
Comment (Alarantalara) [inline src/desktop/apple_notification.mm]: As you're also dropping 10.7 support, this test is no longer needed (it's testing to confirm the OS has the NSUserNotificationCenter API and 10.8 and later always have that).

Comment (hrubymar10) [reply src/desktop/apple_notification.mm]: Sure, that's why I complete removed `HAVE_NS_USER_NOTIFICATION` macro

Comment (Alarantalara) [reply src/desktop/apple_notification.mm]: The Macro confirms the compiler has support for the class (it prevents compile errors). This is the matching runtime test, which can also be eliminated once the binary no longer runs on 10.7

**Pull request merged**